src/resources/SchemaService/SchemaServiceInterfaces.ts

```diff
@@ -1,27 +1,34 @@
 import {SourceModel, CreateSourceModel, CreateSourceOptions} from '../Sources/SourcesInterfaces';
 
-export interface SchemaEntities {
-    entities: SimpleSchemaEntity[];
+export interface SimpleSchemaEntity {
+    id: string;
+    displayName: string;
 }
 
-export interface SchemaEntity extends SimpleSchemaEntity {
-    fields?: SchemaField[];
-    recordCount?: number;
+export interface SimpleSchemaEntityMap {
+    [index: string]: SimpleSchemaEntity;
 }
 
-export interface SimpleSchemaEntity {
-    id: string;
-    displayName: string;
+export interface SchemaEntities {
+    entities: SimpleSchemaEntityMap;
```

Comment:
```suggestion
    entities: Record<string, SimpleSchemaEntity>;
```

And you can remove `SimpleSchemaEntityMap`